controllers/errors.go

```diff
@@ -0,0 +1,30 @@
+package controllers
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/types"
+)
+
+type migrationError interface {
+	error
+
+	migrationName() types.NamespacedName
+}
+
+type migrationErrorStruct struct {
+	message              string
```

Comment:
I think `go vet` might not complain if we make this an error and use `fmt.Errorf` instead of `fmt.Sprintf`.